src/java.base/share/classes/java/nio/channels/Channels.java

```diff
@@ -541,7 +542,7 @@ public static Writer newWriter(WritableByteChannel ch,
                                    int minBufferCap)
     {
         Objects.requireNonNull(ch, "ch");
-        return StreamEncoder.forEncoder(ch, enc.reset(), minBufferCap);
```

Comment:
You could avoid one level of indirection by using `StreamEncoder.forOutputStreamWriter` instead of `new OutputStreamWriter` here. `OutputStreamWriter` is a pass-through wrapper for `StreamEncoder.forOutputStreamWriter` (except for [JDK-6856817](https://bugs.openjdk.org/browse/JDK-6856817) / dc93a924ee4d3b53f0271af577148f8a53eedb57, but that's another story)

Reply:
So changed in c225b9fae846a673d8c372509aa90cd2ead8a3fd. Thanks.